### PR TITLE
Added generation of a DeleteOperation when generating resource

### PR DESCRIPTION
**Pull request by nodeMD**
## Purpose
https://github.com/luckyframework/lucky/issues/1499

## Description
Generating a resource will also generate a DeleteOperation.

I decided to name the generated operation files like: delete_{{underscored_resource}} and save_{{underscored_resource}} and then I have checked that `operation_filename` became not used anywhere so I decided to remove it. Please guide me if that was a proper decision.

## Checklist
* [x] - An issue already exists detailing the issue/or feature request that this PR fixes
* [x] - All specs are formatted with `crystal tool format spec src`
* [ ] - Inline documentation has been added and/or updated
* [x] - Lucky builds on docker with `./script/setup`
* [x] - All builds and specs pass on docker with `./script/test`

### CODE REVIEW

File: tasks/gen/resource/browser.cr [hunk 1]
```diff
@@ -148,13 +149,11 @@ class Lucky::ResourceTemplate < Teeplate::FileTree
...
 
   getter resource, columns
-  getter operation_filename : String,
-    query_filename : String,
+  getter query_filename : String,
     underscored_resource : String,
     folder_name : String
 
   def initialize(@resource : String, @columns : Array(Lucky::GeneratedColumn))
-    @operation_filename = operation_class.underscore
```
Comment (jwoertink) [inline tasks/gen/resource/browser.cr:157]: It looks like the `operation_class` method on line 174 can be removed.

File: tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Delete<%= resource %> < <%= resource %>::DeleteOperation
+  # Read more on deleting records
+  # https://luckyframework.org/guides/database/deleting-records
+end
```
Comment (jwoertink) [inline tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr:4]: Can you add a linebreak after this?

File: tasks/gen/resource/browser.cr [hunk 3]
```diff
@@ -103,6 +103,7 @@ class Gen::Resource::Browser < LuckyTask::Task
   private def display_success_messages
     success_message(resource_name, "./src/models/#{underscored_resource}.cr")
     success_message("Save" + resource_name, "./src/operations/save_#{underscored_resource}.cr")
+    success_message("Delete" + resource_name, "./src/operations/delete_#{underscored_resource}.cr")
```
Comment (jwoertink) [inline tasks/gen/resource/browser.cr:106]: Now that this file is generated, we just need to fix your previous PR 😄 

https://github.com/luckyframework/lucky/blob/master/tasks/gen/templates/resource/actions/%7B%7Bfolder_name%7D%7D/delete.cr.ecr#L4

This file should change from `<%= resource %>::DeleteOperation` to `Delete<%= resource %>`

**Review by jwoertink — CHANGES_REQUESTED** [review]
Awesome! Another great contribution. Just a few minor changes, and this one will be good 👍  Thanks again for doing this.

File: tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr (hunk 2 above)
Comment (nodeMD) [reply tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr:4]: Sure 👍 Why is it important to have an empty line at the end of a file?

File: tasks/gen/resource/browser.cr (hunk 1 above)
Comment (nodeMD) [reply tasks/gen/resource/browser.cr:157]: I see that it is used in other resource ecr's. I will remove it and replace the occurrences with Save<%= resource %>

File: tasks/gen/resource/browser.cr (hunk 3 above)
Comment (nodeMD) [reply tasks/gen/resource/browser.cr:106]: Aaaah now I see it 😄 thanks!

File: tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr (hunk 2 above)
Comment (jwoertink) [reply tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr:4]: We make sure that all generated code will pass the Ameba tests by default. Even if a project doesn't use Ameba, we still want to make sure they would pass so if someone does add Ameba in, they're not having to fix issues like this.

Comment (nodeMD) [reply tasks/gen/templates/resource/operations/delete_{{underscored_resource}}.cr.ecr]: got it, thank you! 🙇

**Review by jwoertink — APPROVED** [review]
This is awesome! Thank you 🎉

**Pull request merged**